### ISSUE TITLE
google_compute_security_policy very slow

### ISSUE THREAD

**Issue by fawaf** [hashicorp/terraform-provider-google#18466]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to a user, that user is claiming responsibility for the issue.
* Customers working with a Google Technical Account Manager or Customer Engineer can ask them to [reach out internally](https://github.com/hashicorp/terraform-provider-google/wiki/Customer-Contact#raising-gcp-internal-issues-with-the-provider-development-team) to expedite investigation and resolution of this issue.

### Terraform Version & Provider Version(s)

Terraform v1.5.7
on linux_amd64
+ provider registry.terraform.io/hashicorp/google v5.34.0
+ provider registry.terraform.io/hashicorp/http v3.4.3

### Affected Resource(s)

google_compute_security_policy

### Terraform Configuration

```tf

resource "google_compute_security_policy" "blah" {
  name        = "${var._cell_short_name}-blah-policy"
  description = "Security policy for monolith proxy"
  project     = var._gcp_project_id

  lifecycle {
    ignore_changes = [
      rule,
    ]
  }
}
```

### Debug Output

https://gist.github.com/fawaf/88a8b9a519ce60378a92575dfee56021

### Expected Behavior

run to completion

### Actual Behavior

hangs on pulling all rules

### Steps to reproduce

1. `terraform apply`

**Comment by fawaf** [comment]
oh, forgot to mention that the security policy itself contains 2000 rules already. basically, running tf on an and existing security policy causes this issue.

**Comment by fawaf** [comment]
updated ticket description to reflect that.

**Comment by ggtisc** [comment]
Well this is definitively the issue, you could decouple your policy in different rules if you want to have a faster creation or wait until it finishes.

**Comment by fawaf** [comment]
it succeeded a few times before, but then just started hanging after a few usages. we also are running into a quota issue in that project. i'll report back once we get that solved.

**Comment by ggtisc** [comment]
Yes, this is because you are overcharging the policy. For this reason it is a good practice to decouple your resources as much as possible when you work in cloud engineering. With this you ensure to have a clear code, it is easier to give maintenance, reuse your code (in this case if you need to reuse some policy rules) and avoid overcharge API calls. 

In the end you could do that, but this increases the timeout.
